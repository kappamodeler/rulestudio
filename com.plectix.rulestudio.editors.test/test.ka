# test file
'rule' A(x) -> B(y)
%init: 1 * a(s1!1,s2~u),b(s1!1)
