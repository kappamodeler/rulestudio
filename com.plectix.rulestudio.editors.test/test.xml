<?xml version="1.0" encoding="utf-8"?>
	<ComplxSession xmlns="http://plectix.synthesisstudios.com/schemas/kappasession" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" CommandLine="--no-build-influence-map --no-compute-qualitative-compression --no-compute-quantitative-compression --no-dump-iteration-number --no-dump-rule-iteration --no-enumerate-complexes --contact-map C:\Users\bill\plectix\svn\eclipse_plugins\com.plectix.kappa.editors.test\test.ka --xml-session-name C:\Users\bill\plectix\svn\eclipse_plugins\com.plectix.kappa.editors.test\test.xml" InputFile="C:\Users\bill\plectix\svn\eclipse_plugins\com.plectix.kappa.editors.test\test.ka" TimeStamp="2009-12-29 21:16:00">
		<RuleSet Name="Original">
			<Rule Type="RULE" Name="rule" Id="1" Data="A(x)-&gt;B(y)">
			</Rule>
		</RuleSet>
		<ContactMap Name="High resolution">
			<Agent Name="a">
				<Site Name="s1" CanChangeState="false" CanBeBound="true">
				</Site>
				<Site Name="s2" CanChangeState="true" CanBeBound="false">
				</Site>
			</Agent>
			<Agent Name="b">
				<Site Name="s1" CanChangeState="false" CanBeBound="true">
				</Site>
			</Agent>
			<Bond FromAgent="a" FromSite="s1" ToAgent="b" ToSite="s1">
			</Bond>
			<Bond FromAgent="b" FromSite="s1" ToAgent="a" ToSite="s1">
			</Bond>
		</ContactMap>
		<Log>
			<Entry Position="1262139360.378" Count="1" Message="JSIM: Build on 2009-12-07 16:00:02 EST from Revision SVN 10348, JRE: Sun Microsystems Inc. 1.6.0_17" Type="INFO">
			</Entry>
			<Entry Position="1262139360.378" Count="1" Message="-Initialization..." Type="INFO">
			</Entry>
			<Entry Position="1262139360.380" Count="1" Message="--Computing initial state" Type="INFO">
			</Entry>
			<Entry Position="1262139360.381" Count="1" Message="-Initialization: 0.0 sec. CPU" Type="INFO">
			</Entry>
		</Log>
	</ComplxSession>