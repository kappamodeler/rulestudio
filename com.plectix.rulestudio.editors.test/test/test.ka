#
# Welcome to the Kappa Workbench Editor 
# 
# This editor will improve the speed of entering valid kappa by using auto completion and basic syntax checking as you write each line.
# Auto completion of agents, rules and Kappa will drop down when you are creating initial conditions, observables or rules that reuse agents in prior rules.
# 
# Please send comments and feedback on our editor to eclipse@cellucidate.com
#

# Check ideas
a()->b(s)
a(s)<-b(s)



#INVALID Rules
a(s)->a(s!_)
a(s)->a(s?)
a(),b(s?)->a(),b(s)
a(s!_) <-> a(s)
B(y~a~e)->B(y~a~e)
agent(x),B(y~a?) -> agent(x!1),B(y~a!1)
Aww(x) -> Aww(x?)
A(x!_) <-> A(x)
A(x) -> A(x!_)
A(x!_),B(y) <-> A(x!1),B(y!1)
A(x),B(y),C(v),D(x) -> A(x!1),B(y!1),C(v!1),D(x!1)
A(x),B(y),C(v),D(x) -> A(x!1),B(y!1),C(v!1),D(x)
A(x~u) -> A(x)
A(x) -> A(x~u)
A(x~u),B(x) -> A(x!1),B(x!1)
A(x!_) -> A(x~a)
A(x~u!_) -> A(x!_)
A(x?) -> A(x~u?)
A(x~a) -> A(~u)
A(~a) -> A(x~u)
A(x,y~a) -> A(x,~u)
A(x~a),B(x) <-> A(~u!1),B(x)
a(s1!1),b(s1)->
->a(s1!1),b(s1)
a(s1!1),b(s1!1),c(s1!1)->
->a(s1!1),b(s1!1),c(s1!1)
a(s1!1,s2!1)->
->a(s1!1,s2!1)
a(s1,s2,s3)->a(s1,s2)
a(s)->a(s!_)
a(s)->a(s?)
a(),b(s?)->a(),b(s)

'E_Sy_op' E(s!0), S(s~y) -> E(s!0), S(s~y)
'E_Sn' E(s), S(s~n) <- E(s!0), S(s~n!0)
'E_Sn' E(s!0), S(s~n!0) ->  E(s), S(s~n)
'E_Sn' E(s), S(s~n) <-> E(s!0), S(s!0)

#VALID rules
a(s!_)->a(s)
B(y~a?) -> B(y~b?)
Aww(x?) ->
Aww(x?) -> B(y) 
A(x!_) -> A(x)
A(x!_),B(y) -> A(x!1),B(y!1)
A(x),B(y),C(v),D(x) -> A(x!1),B(y!1),C(v!2),D(x!2)
A(x~u) -> A(x~p)
A(x~u),B(x) -> A(x~a!1),B(x!1)
A(x~u!_) -> A(x~a)
A(x~u!_) -> A(x~a!_)
A(x~u?) -> A(x~a?)
A(x~a) -> A(x~u)
A(x~a) -> A(x~u)
A(x,y~a) -> A(x,y~u)
A(x~a),B(x) <-> A(x~u!1),B(x!1)
a(s1!1,s2~u),b(s1!1)->a(s1,s2~p),b(s1)
a(s1,s2,s3)->a(s3,s2,s1)
A(x~a) -> A(x~u) @ 6.345E-1
A(x~a) -> A(x~u) @ 6.345E+1
A(x~a) -> A(x~u) @ 6.345E1
A(x~a) -> A(x~u) @ 6.345e-1
A(x~a) -> A(x~u) @ 6E-1


#INVALID agents
%init: 1 * a(s1,s1)
%init: 1 * abcdefghijklmnopqrstu()
%init: 1 * a(abcdefghijklmnopqrstu)
%init: 1 * a(s~abcdefghijklmnopqrstu)
%init: 1 * (s)
%init: 1 * a(,s)
%init: 1 * &()
%init: 1 * a#()
%init: 1 * a$()
%init: 1 * ^a()
%init: 1 * -a()
%init: 1 * _a()
%init: 1 * a(s%)
%init: 1 * a(s*)
%init: 1 * a(s@)
%init: 1 * a(^1)
%init: 1 * a(-b)
%init: 1 * a(_c)
%init: 1 * a(s~^)
%init: 1 * a(s~-)
%init: 1 * a(s~_)
%init: 1 * a(s1!1),b(s1)
%init: 1 * a(s1!1),b(s1!1),c(s1!1)
%init: 1 * a(s1!1,s2!1)
%init: 1 * a(s!_) #ICs can't contain any-bonds 
%init: 1 * a(s?) #ICs can't contain wild-card bonds 
%init: 1 * B(y~a~e) #sites in ICs can't have multiple states 
%init: 1 * A(x!1),B(y!1),C(v!1),D(x!1) #bond in ICs have to appear twice only 
%init: 1 * A(x!1),B(y!1),C(v!1),D(x) #bond in ICs have to appear twice only 
%init: 1 * A(x~u!_) #ICs can't contain any-bonds 
%init: 1 * A(x~u?) #ICs can't contain wild-card bonds 
%init: 1 * A(~u) #can't have state with no corresponding site 
%init: 1 * A(~u!1),B(x!1) #can't have state with no corresponding site 
%init: 1 * a(s1!1),b(s1) #bond in ICs have to appear twice only 
%init: 1 * a(s1!1,s2!1) #bond in ICs can't link two sites in same agent 

#VALID
%init: 1 * a(s1,s2~u)
%init: 1 * azAZ09-_^()
%init: 1 * a(azAZ09-_^)
%init: 1 * a(s~azAZ09)
%init: 1 * a(s1!1,s2~u),b(s1!1)



#INVALID FILE
#'invalid rule - Del Del Add Add Bnd' A(x),B(x) -> B(x!1),A(x!1)
#%init: 100 * A(x,y),B(x,y)
#
#INVALID FILE
#'invalid rule - uses site not mentioned in ICs' A(xx!_) -> A(xx)
#%init: 100 * A(x,y),B(x,y)
#
#VALID FILE
#'valid (but stupid) rule' A(x,y),B(x,y) -> B(x!1,y),A(x!1,y)
#%init: 100 * A(x,y),B(x,y)
