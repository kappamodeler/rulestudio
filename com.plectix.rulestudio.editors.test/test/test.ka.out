# Created Mon Dec 28 17:01:06 EST 2009
# start	end	level	message
Line 15, Char 1	463	475	Warning	The any-bond (ie, "!_") can appear on both sides of a rule on the same sites. However, the only circumstance where it can appear only on one side of a rule is when it is on the left hand side of a non-reversible rule (ie, a rule with a forward arrow only).
Line 16, Char 1	476	487	Warning	Wild-card bonds (ie, "?") must appear on both side of the rule (on the same site). The only exception to this is when the agent is degraded (ie, appears on the left hand side but not the right hand side of the rule) - in this case the sites on the agent on the LHS can have wild-card bonds.
Line 17, Char 1	488	507	Warning	Wild-card bonds (ie, "?") must appear on both side of the rule (on the same site). The only exception to this is when the agent is degraded (ie, appears on the left hand side but not the right hand side of the rule) - in this case the sites on the agent on the LHS can have wild-card bonds.
Line 18, Char 1	508	523	Warning	The any-bond (ie, "!_") can appear on both sides of a rule on the same sites. However, the only circumstance where it can appear only on one side of a rule is when it is on the left hand side of a non-reversible rule (ie, a rule with a forward arrow only).
Line 19, Char 3	526	528	Warning	The State "a~e" is not valid needs to follow the following form ([a-z A-Z 0-9])*.
Line 19, Char 13	536	538	Warning	The State "a~e" is not valid needs to follow the following form ([a-z A-Z 0-9])*.
Line 20, Char 1	543	582	Warning	Wild-card bonds (ie, "?") must appear on both side of the rule (on the same site). The only exception to this is when the agent is degraded (ie, appears on the left hand side but not the right hand side of the rule) - in this case the sites on the agent on the LHS can have wild-card bonds.
Line 21, Char 1	583	600	Warning	Wild-card bonds (ie, "?") must appear on both side of the rule (on the same site). The only exception to this is when the agent is degraded (ie, appears on the left hand side but not the right hand side of the rule) - in this case the sites on the agent on the LHS can have wild-card bonds.
Line 22, Char 1	601	616	Warning	The any-bond (ie, "!_") can appear on both sides of a rule on the same sites. However, the only circumstance where it can appear only on one side of a rule is when it is on the left hand side of a non-reversible rule (ie, a rule with a forward arrow only).
Line 23, Char 1	617	631	Warning	The any-bond (ie, "!_") can appear on both sides of a rule on the same sites. However, the only circumstance where it can appear only on one side of a rule is when it is on the left hand side of a non-reversible rule (ie, a rule with a forward arrow only).
Line 24, Char 1	632	661	Warning	The any-bond (ie, "!_") can appear on both sides of a rule on the same sites. However, the only circumstance where it can appear only on one side of a rule is when it is on the left hand side of a non-reversible rule (ie, a rule with a forward arrow only).
Line 25, Char 1	662	712	Warning	This rule has "4" instances of the bond "1" on the right hand side.
Line 26, Char 1	713	761	Warning	This rule has "3" instances of the bond "1" on the right hand side.
Line 27, Char 1	762	776	Warning	The site "x" has a state on one side of a rule, and not on the other.
Line 28, Char 1	777	791	Warning	The site "x" has a state on one side of a rule, and not on the other.
Line 29, Char 1	792	820	Warning	The site "x" has a state on one side of a rule, and not on the other.
Line 30, Char 1	821	837	Warning	The site "x" has a state on one side of a rule, and not on the other.
Line 31, Char 1	838	856	Warning	The site "x" has a state on one side of a rule, and not on the other.
Line 31, Char 1	838	856	Warning	The any-bond (ie, "!_") can appear on both sides of a rule on the same sites. However, the only circumstance where it can appear only on one side of a rule is when it is on the left hand side of a non-reversible rule (ie, a rule with a forward arrow only).
Line 32, Char 1	857	873	Warning	The site "x" has a state on one side of a rule, and not on the other.
Line 33, Char 11	884	885	Warning	State (u) must be preceded by site.
Line 33, Char 1	874	889	Warning	The site "x" has a state on one side of a rule, and not on the other.
Line 34, Char 1	890	891	Warning	State (a) must be preceded by site.
Line 34, Char 1	890	905	Warning	The site "x" has a state on one side of a rule, and not on the other.
Line 35, Char 13	918	919	Warning	State (u) must be preceded by site.
Line 35, Char 1	906	925	Warning	The site "y" has a state on one side of a rule, and not on the other.
Line 36, Char 19	944	945	Warning	State (u) must be preceded by site.
Line 36, Char 1	926	954	Warning	The site "x" has a state on one side of a rule, and not on the other.
Line 36, Char 1	926	954	Warning	This rule has "1" instances of the bond "1" on the right hand side.
Line 37, Char 1	955	970	Warning	This rule has "1" instances of the bond "1" on the left hand side.
Line 38, Char 1	971	986	Warning	This rule has "1" instances of the bond "1" on the right hand side.
Line 38, Char 1	971	986	Error	The line of kappa isn't valid.
Line 39, Char 1	987	1012	Warning	This rule has "3" instances of the bond "1" on the left hand side.
Line 40, Char 1	1013	1038	Warning	This rule has "3" instances of the bond "1" on the right hand side.
Line 40, Char 1	1013	1038	Error	The line of kappa isn't valid.
Line 42, Char 1	1054	1068	Error	The line of kappa isn't valid.
Line 44, Char 1	1091	1103	Warning	The any-bond (ie, "!_") can appear on both sides of a rule on the same sites. However, the only circumstance where it can appear only on one side of a rule is when it is on the left hand side of a non-reversible rule (ie, a rule with a forward arrow only).
Line 45, Char 1	1104	1115	Warning	Wild-card bonds (ie, "?") must appear on both side of the rule (on the same site). The only exception to this is when the agent is degraded (ie, appears on the left hand side but not the right hand side of the rule) - in this case the sites on the agent on the LHS can have wild-card bonds.
Line 46, Char 1	1116	1135	Warning	Wild-card bonds (ie, "?") must appear on both side of the rule (on the same site). The only exception to this is when the agent is degraded (ie, appears on the left hand side but not the right hand side of the rule) - in this case the sites on the agent on the LHS can have wild-card bonds.
